includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -676,6 +676,10 @@ public function remove_all_actions_on_preflight_check( $response, $handler, $req
 			remove_all_actions( 'woocommerce_store_api_checkout_order_processed' );
 			// Avoid increasing coupon usage count during preflight check.
 			remove_all_actions( 'woocommerce_order_status_pending' );
+
+			// Avoid creating new accounts during preflight check.
+			remove_all_filters( 'woocommerce_checkout_registration_enabled' );
```

Comment:
Do we need to remove this filter? I don't have issue checking out without removing it.

Reply:
Fixed on https://github.com/Automattic/woocommerce-payments/pull/8924/commits/a092fc94492e0480810a188a05185646c661524a.